### ISSUE TITLE
Should Style/NegatedIfElseCondition be able to allow inequality operators?

### ISSUE THREAD

**Issue by tsugimoto** [rubocop/rubocop#10925]
## Is your feature request related to a problem? Please describe.

In cases when `Style/NumericPredicate` is `Enabled: false` or `EnforcedStyle: comparison`, the branches would get swapped by `Style/NegatedIfElseCondition`. 

```rb
# frozen_string_literal: true

result = user.coins != 0 ? :have_coins : :no_coins
puts result
```

```
Inspecting 1 file
C

Offenses:

main.rb:3:10: C: [Correctable] Style/NegatedIfElseCondition: Invert the negated condition and swap the ternary branches.
result = user.coins != 0 ? :have_coins : :no_coins
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

1 file inspected, 1 offense detected, 1 offense autocorrectable
```

Correcting the offensing line into `result = user.coins == 0 ? :no_coins : :have_coins` sometimes feels awkward.

In this particular case:
- The swapped code reads like "If the user does not have any coins then `:no_coins`, else `:have_coins`."
- This swap happens anyway if `Style/NumericPredicate` is `predicate`, but one may opt to `non_zero?` in that case.

## Describe the solution you'd like

Add a configuration to `Style/NegatedIfElseCondition` Cop to look for prefix `!`s but overlook `!=`s, possibly named `AllowComparison: true` or `AllowInequality: true`.

## Describe alternatives you've considered

- Disable `Style/NegatedIfElseCondition` Cop entirely, although I would like prefix `!`s to be flagged.
- Exclude `!=` from the `Style/NegatedIfElseCondition` Cop. However this is debatable.

## Additional context

What about pattern matching operators `!~`? (I think they should be flagged)
```
$ rubocop -V
1.35.0 (using Parser 3.1.2.1, rubocop-ast 1.21.0, running on ruby 3.1.1 x86_64-darwin20)
```
